platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/event/tipset/TipsetEventCreatorTests.java

```diff
@@ -1096,4 +833,75 @@ void checkSettingEventBirthRound(final boolean advancingClock, final boolean use
+
+        final List<PlatformEvent> pcesEvents = new ArrayList<>();
+        PlatformEvent eventWithHighestNGen = null;
+        for (int i = 0; i < numEvents; i++) {
+            final PlatformEvent event = createTestEvent(random, selfId, i++);
+            if (eventWithHighestNGen == null || event.getNGen() > eventWithHighestNGen.getNGen()) {
+                eventWithHighestNGen = event;
+            }
+            pcesEvents.add(event);
+        }
+
+        // Add the events to the creator in a random order
+        Collections.shuffle(pcesEvents, random);
+        pcesEvents.forEach(eventCreator::registerEvent);
+
+        // Verify that the new event created uses a self parent that is the event with the highest nGen.
+        // This new event should not have an nGen assigned.
+        final PlatformEvent newEvent = eventCreator.maybeCreateEvent();
+        assertNotNull(newEvent);
+        assertEquals(eventWithHighestNGen.getDescriptor(), newEvent.getSelfParent());
+        assertEquals(EventConstants.GENERATION_UNDEFINED, newEvent.getNGen());
+    }
+
+    /**
+     * This test verifies that an event recently created by the event creator is not overwritten when it learns of a
+     * self event for the first time from the intake pipeline.
+     */
+    @Test
+    @DisplayName("Last Self Event just created is not overwritten")
+    void lastSelfEventNotOverwritten() {
+        final Random random = getRandomPrintSeed();
+        final int networkSize = 1;
+        final Roster roster =
+                RandomRosterBuilder.create(random).withSize(networkSize).build();
+        final NodeId selfId = NodeId.of(roster.rosterEntries().getFirst().nodeId());
+        final EventCreator eventCreator =
+                buildEventCreator(random, new FakeTime(), roster, selfId, () -> Collections.EMPTY_LIST);
+
+        final PlatformEvent newEvent = eventCreator.maybeCreateEvent();
+        assertNotNull(newEvent);
+        assertEquals(EventConstants.GENERATION_UNDEFINED, newEvent.getNGen());
+
+        // Create a self event with an nGen value set and register it with the event creator. This can happen
+        // if we are forced to reconnect and learn of an event we created a long time ago after we started creating
+        // the events. This is a branch, but not necessarily an intentional branch. This old event should be discarded
+        // because we want to favor any self event last created by the event creator even though it does not have an
+        // nGen set.
+        final PlatformEvent oldSelfEvent = createTestEvent(random, selfId, EventConstants.FIRST_GENERATION);
+        eventCreator.registerEvent(oldSelfEvent);
+        final PlatformEvent newEvent2 = eventCreator.maybeCreateEvent();
+        assertNotNull(newEvent);
+        assertEquals(newEvent.getDescriptor(), newEvent2.getSelfParent());
+        assertEquals(EventConstants.GENERATION_UNDEFINED, newEvent.getNGen());
```

Comment:
Copy & paste error? I assume you want to check `newEvent2` is not `null` and has an undefined nGen.